documentation/specs/src/interoperability/ethereum-bridge/ethereum_events_attestation.md

```diff
@@ -54,104 +54,93 @@ keys involved are:
 /eth_msgs/\$msg_hash/seen: bool
 ```
 
-`\$msg_hash` is the SHA256 digest of the Borsh serialization of the relevant 
+`\$msg_hash` is the SHA256 digest of the Borsh serialization of the relevant
 `EthereumEvent`.
 
-Changes to this `/eth_msgs` storage subspace are only ever made by 
-nodes as part of the ledger code based on the aggregate of vote 
-extensions for the last Tendermint round. That is, changes to `/eth_msgs` happen 
-in block `n+1` in a deterministic manner based on the vote extensions of the 
-Tendermint round for block `n`. The `/eth_msgs` storage subspace will belong 
-to the `EthBridge` validity predicate. It should disallow any changes to 
+Changes to this `/eth_msgs` storage subspace are only ever made by
+nodes as part of the ledger code based on the aggregate of votes
+by validators for specific events. That is, changes to
+`/eth_msgs` happen
+in block `n` in a deterministic manner based on the votes included in the
+block proposal for block `n`. Depending on the underlying Tendermint
+version, these votes will either be included as vote extensions or as
+protocol transactions.
+
+The `/eth_msgs` storage subspace will belong
+to the `EthBridge` validity predicate. It should disallow any changes to
 this storage from wasm transactions.
 
-## Including events into storage
+### Including events into storage
 
-For every Namada block proposal, the vote extension of a validator should include
-the events of the Ethereum blocks they have seen via their full node such that:
-1. The storage value `/eth_msgs/\$msg_hash/seen_by` does not include their
-   address.
-2. It's correctly formatted.
-3. It's reached the required number of confirmations on the Ethereum chain
+For every Namada block proposal, block proposer should include the votes for
+events from other validators into their proposal. If the underlying Tendermint
+version supports vote extensions, consensus invariants guarantee that a
+quorum of votes from the previous block height can be included. Otherwise,
+validators can only submit votes by broadcasting protocol transactions,
+which comes with less guarantees.
 
-Each event that a validator is voting to include must be individually signed by 
+The vote of a validator should include the events of the Ethereum blocks they
+have seen via their full node such that:
+1. It's correctly formatted.
+2. It's reached the required number of confirmations on the Ethereum chain
+
+Each event that a validator is voting to include must be individually signed by
 them. If the validator is not voting to include any events, they must still
-provide a signed voted extension indicating this.
+provide a signed empty vector of events to indicate this.
```

Comment:
mention that we sign block heights as a way to prevent replay attacks? or do we already mention this elsewhere?